### DISCUSSION TITLE
Best Practices for Publishing a Turbo Module and Fabric Component

### DISCUSSION THREAD

**Discussion by jerielng** [reactwg/react-native-new-architecture#248] [Q&A] [answered]
I'm working on building out a backwards-compatible Fabric Component, which depends on an existing backwards-compatible Turbo Module. Ideally, I'd like to publish both of these as part of the same library, but I'm looking to understand what the best practices are for structuring a project like this.

From what I understand, it's no longer possible to add an array of specs to the `codegenConfig` in a single `package.json`, so is my understanding correct that we would need to maintain separate `package.json` files as well as namespaces for each module or component?

**Comment by cortinico** [comment, marked as answer]
> From what I understand, it's no longer possible to add an array of specs to the codegenConfig in a single package.json, so is my understanding correct that we would need to maintain separate package.json files as well as namespaces for each module or component?

Correct. You can place all the specs in the same folder, and they all will be generated.
Otherwise, you can create separate NPM packages with distinct spec folders.
